File: app/src/main/java/com/desafioconcrete/jonasmferreira/appgitjavapop/adapter/PullRequestAdapter.java
package com.desafioconcrete.jonasmferreira.appgitjavapop.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.desafioconcrete.jonasmferreira.appgitjavapop.R;
import com.desafioconcrete.jonasmferreira.appgitjavapop.domain.PullRequest;
import com.desafioconcrete.jonasmferreira.appgitjavapop.utils.Tools;
import com.nostra13.universalimageloader.core.ImageLoader;

import java.util.ArrayList;
import java.util.List;

public class PullRequestAdapter extends RecyclerView.Adapter<PullRequestAdapter.ViewHolder>{
    List<PullRequest> pullrequestList = new ArrayList<>();

    public PullRequestAdapter(List<PullRequest> pullrequestList) {
        this.pullrequestList = pullrequestList;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);

        View pullrequestView = inflater.inflate(R.layout.item_pull_request, parent, false);

        ViewHolder viewHolder = new ViewHolder(pullrequestView);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {

        ImageLoader imageLoader = ImageLoader.getInstance();

        PullRequest pullrequest = pullrequestList.get(position);

        TextView tvPullRequestTitle = holder.tvPullRequestTitle;
        tvPullRequestTitle.setText(pullrequest.getTitle());

        TextView tvPullRequestDescr = holder.tvPullRequestDescr;
        tvPullRequestDescr.setText(pullrequest.getBody());

        TextView tvPullRequestUsername = holder.tvPullRequestUsername;
        tvPullRequestUsername.setText(pullrequest.getUser().getLogin());

        TextView tvPullRequestFullname = holder.tvPullRequestFullname;
        tvPullRequestFullname.setText(pullrequest.getUser().getLogin());

        TextView tvPullRequestDate = holder.tvPullRequestDate;
        String dt = pullrequest.getCreatedAt();
        dt = dt.replace("T"," ");
        dt = dt.replace("Z","s");
        tvPullRequestDate.setText(Tools.dateDB2BR(dt.trim()));

        ImageView civPullRequestUser = holder.civPullRequestUser;

        imageLoader.displayImage(pullrequest.getUser().getAvatarUrl(), civPullRequestUser);

    }

    @Override
    public int getItemCount() {
        return pullrequestList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public TextView tvPullRequestTitle,tvPullRequestDescr,tvPullRequestUsername,tvPullRequestFullname,tvPullRequestDate;
        public ImageView civPullRequestUser;
        public ViewHolder(View itemView) {
            super(itemView);
            tvPullRequestTitle = (TextView) itemView.findViewById(R.id.tvPullRequestTitle);
            tvPullRequestDescr = (TextView) itemView.findViewById(R.id.tvPullRequestDescr);
            tvPullRequestUsername = (TextView) itemView.findViewById(R.id.tvPullRequestUsername);
            tvPullRequestFullname = (TextView) itemView.findViewById(R.id.tvPullRequestFullname);
            civPullRequestUser = (ImageView)itemView.findViewById(R.id.civPullRequestUser);
            tvPullRequestDate = (TextView) itemView.findViewById(R.id.tvPullRequestDate);
        }
    }
}

File: app/src/main/java/com/example/lucasdias/mvvmpoc/presentation/ui/pullrequests/PullRequestAdapter.kt
package com.example.lucasdias.mvvmpoc.presentation.ui.pullrequests

import android.support.v7.widget.RecyclerView
import android.support.v7.widget.RecyclerView.Adapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.example.lucasdias.mvvmpoc.R
import com.example.lucasdias.mvvmpoc.domain.entity.PullRequest
import com.example.lucasdias.mvvmpoc.presentation.extensions.loadUrl
import timber.log.Timber

class PullRequestAdapter(var view: PullRequestActivity): Adapter<PullRequestAdapter.ViewHolder>() {

    private var pullRequestList = ArrayList<PullRequest>()

    fun updatePullRequestList(pullRequest: ArrayList<PullRequest>) {
        this.pullRequestList = pullRequest
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val viewLayout = LayoutInflater.from(parent.context).inflate(R.layout.pull_request_list_item, parent, false)
        return ViewHolder(viewLayout)
    }

    override fun getItemCount(): Int {
        return pullRequestList.count()    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        if (pullRequestList.isNotEmpty()) {
            holder.bind(pullRequestList[position], view)
        }
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val title by lazy { itemView.findViewById<TextView>(R.id.pullRequest_tvPullRequestTitle) }
        private val description by lazy { itemView.findViewById<TextView>(R.id.pullRequest_tvPullRequestDescription) }
        private val createdAt by lazy { itemView.findViewById<TextView>(R.id.pullRequest_tvPullRequestCreatedAt) }
        private val userName by lazy { itemView.findViewById<TextView>(R.id.pullRequest_tvPullRequestUserName) }
        private val userPicture by lazy { itemView.findViewById<ImageView>(R.id.pullRequest_ivPullRequestUserPicture) }

        fun bind(pullRequest: PullRequest, view: PullRequestActivity) {
            title.text = pullRequest.title
            description.text = pullRequest.description
            createdAt.text = pullRequest.getCreatedAtDateString()
            userName.text = pullRequest.user?.login
            userPicture.loadUrl(pullRequest.user?.avatarUrl)
            itemView.setOnClickListener {
                if(pullRequest.htmlUrl != ""){
                    Timber.i("PullRequest clicked.")
                    view.callPullRequestOnBrowser(pullRequest.htmlUrl ?: "https://www.google.com")
                }
            }
        }
    }
}